tests/queries/0_stateless/02151_hash_table_sizes_stats_distributed.sh

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env bash
+# Tags: long, distributed
+
+unset CLICKHOUSE_LOG_COMMENT
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+. "$CURDIR"/../shell_config.sh
+
+
+table_size=10000
+max_threads=5
+
+
+prepare_table() {
+  table_name="t_hash_table_sizes_stats_$RANDOM$RANDOM"
+  $CLICKHOUSE_CLIENT -q "DROP TABLE IF EXISTS $table_name;"
+  if [ -z "$1" ]; then
+    $CLICKHOUSE_CLIENT -q "CREATE TABLE $table_name(number UInt64) Engine=MergeTree() ORDER BY tuple();"
+  else
+    $CLICKHOUSE_CLIENT -q "CREATE TABLE $table_name(number UInt64) Engine=MergeTree() ORDER BY $1;"
+  fi
+  $CLICKHOUSE_CLIENT -q "SYSTEM STOP MERGES $table_name;"
+  for ((i = 1; i <= max_threads; i++)); do
+    cnt=$((table_size / max_threads))
+    from=$(((i - 1) * cnt))
+    $CLICKHOUSE_CLIENT -q "INSERT INTO $table_name SELECT * FROM numbers($from, $cnt);"
+  done
+  $CLICKHOUSE_CLIENT -q "DROP TABLE IF EXISTS ${table_name}_d;"
+  $CLICKHOUSE_CLIENT -q "CREATE TABLE ${table_name}_d AS $table_name ENGINE = Distributed(test_cluster_two_shards, currentDatabase(), $table_name);"
+  table_name="${table_name}_d"
+}
+
+prepare_table_with_sorting_key() {
+  prepare_table "$1"
+}
+
+run_query() {
+  query_id="${CLICKHOUSE_DATABASE}_hash_table_sizes_stats_$RANDOM$RANDOM"
+  $CLICKHOUSE_CLIENT --query_id="$query_id" --multiquery -q \
+    "SET max_block_size = $((table_size / 10));
+     SET merge_tree_min_rows_for_concurrent_read = 1;
+     SET max_untracked_memory = 0;
+     $query"
+}
+
+check_number_of_string_occurrences_on_initiator_satisfies_condition() {
+  $CLICKHOUSE_CLIENT -q "SYSTEM FLUSH LOGS"
+  $CLICKHOUSE_CLIENT                                                                                 \
+    --param_query_id="$query_id"                                                                     \
+    -q "WITH (                                                                                       \
+          SELECT COUNT(message)                                                                      \
+            FROM system.text_log                                                                     \
+           WHERE event_date >= yesterday() AND query_id = {query_id:String} AND message ILIKE '%$1%' \
+        ) AS res                                                                                     \
+        SELECT res $2"
+}
+
+check_number_of_string_occurrences_on_peer_satisfies_condition() {
+  $CLICKHOUSE_CLIENT -q "SYSTEM FLUSH LOGS"
+  $CLICKHOUSE_CLIENT                                                                             \
+    --param_query_id="$query_id"                                                                 \
+    -q "WITH (                                                                                   \
```

Comment:
It should work without newline escaping (inside string literal).

Reply:
done